lib/active_hash/relation.rb

```diff
@@ -71,7 +71,20 @@ def pluck(*column_names)
     def reload
       @records = filter_all_records_by_query_hash
     end
-    
+
+    def order(*options)
+      check_if_method_has_arguments!(:order, options)
+      records = where({})
+      return records if options.blank?
+
+      processed_args = preprocess_order_args(options)
+      candidates = records.dup
+
+      order_by_args!(candidates, processed_args)
+
+      candidates
```

Comment:
This seems to returns an `Array`, but I think it would be better to return an `ActiveHash::Relation` again, so that you can chain it.

To make that work, I think instead of returning the duped & ordered `records`, writing them back to `@records` and returning `self` should do it.

Reply:
@nsommer 
Thanks for your comment.

This method returned an `Array` object when I first implemented as you mentioned.
However, it does return `ActiveHash::Relation` after this change cf.https://github.com/zilkey/active_hash/pull/177/commits/fcfa0ab6edbbead9855f4606c3a0b8c4d99431a2.

```ruby
Country.order("code, id DESC").is_a?(ActiveHash::Relation)
=> true
Country.where(language: 'English').order(id: :desc).is_a?(ActiveHash::Relation)
=> true
```

This returns `ActiveHash::Relation` so that it can be used like following example.
```ruby
Country.order("code, id DESC").where(id: 1)
```
Does this resolve your concern?
Let me know if you need further explanation. Would love to share it.

Thanks!

Reply:
Oh, I see. I was confused because of the local `records` variable which gets a relation assigned (with `where({})`), because there is also a private method `records` which returns the data array. Maybe you can rename the local variable to something like `relation_to_order` or something like that?

Reply:
> I was confused because of the local records variable which gets a relation assigned (with where({})), because there is also a private method records which returns the data array. 

Thank you for your feedback!
This makes a lot of sense to me. `records` was a bit confusing. 
 Renamed `records` to `relation`.

Reply:
Great 👍